domain/application/state/unit_test.go

```diff
@@ -955,3 +959,165 @@ func deptr[T any](v *T) T {
+		return tx.QueryRowContext(ctx, "SELECT uuid FROM unit WHERE name = ?", u.UnitName).Scan(&unitUUID)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	unitUUID, err = s.state.GetUnitUUIDByName(context.Background(), coreunit.Name("666"))
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID,
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	var (
+		obtainedUnitUUID string
+		obtainedVersion  string
+		obtainedArch     string
+	)
+	err = s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		stmt := `
+SELECT unit_uuid,
+       version,
+       name
+FROM unit_agent_version
+INNER JOIN architecture ON unit_agent_version.architecture_id = architecture.id
+WHERE unit_uuid = ?
+	`
+
+		return tx.QueryRowContext(ctx, stmt, unitUUID).Scan(
+			&obtainedUnitUUID,
+			&obtainedVersion,
+			&obtainedArch,
+		)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(obtainedUnitUUID, gc.Equals, unitUUID.String())
+	c.Check(obtainedVersion, gc.Equals, jujuversion.Current.String())
+	c.Check(obtainedArch, gc.Equals, corearch.ARM64)
+}
+
+// TestSetRunningAgentBinaryVersionUpdate asserts setting the initial agent binary
+// version (happy path) and then updating the value.
+func (s *unitStateSuite) TestSetRunningAgentBinaryVersionUpdate(c *gc.C) {
+	u := application.InsertUnitArg{
+		UnitName: "666",
+	}
+	s.createApplication(c, "666", life.Alive, u)
+	var unitUUID coreunit.UUID
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		return tx.QueryRowContext(ctx, "SELECT uuid FROM unit WHERE name = ?", u.UnitName).Scan(&unitUUID)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	unitUUID, err = s.state.GetUnitUUIDByName(context.Background(), coreunit.Name("666"))
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID,
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Assert(err, jc.ErrorIsNil)
+
+	var (
+		obtainedUnitUUID string
+		obtainedVersion  string
+		obtainedArch     string
+	)
+	err = s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		stmt := `
+SELECT unit_uuid,
+       version,
+       name
+FROM unit_agent_version
+INNER JOIN architecture ON unit_agent_version.architecture_id = architecture.id
+WHERE unit_uuid = ?
+	`
+
+		return tx.QueryRowContext(ctx, stmt, unitUUID).Scan(
+			&obtainedUnitUUID,
+			&obtainedVersion,
+			&obtainedArch,
+		)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(obtainedUnitUUID, gc.Equals, unitUUID.String())
+	c.Check(obtainedVersion, gc.Equals, jujuversion.Current.String())
+
+	// Update
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID,
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
```

Comment:
We're not updating anything here - the binary version is being set to the same thing as before.
Use a different version number. 
```
newVers := jujuversion.Current
newVers.Patch = newVers.Patch + 1
```

Also in the checks below, need to check for obtained version as well, not just arch

Reply:
fixed.